app/src/main/java/com/muzima/view/patients/PatientLocationMapActivity.java

```diff
@@ -0,0 +1,295 @@
+            @Override
+            public void run() {
+                getDirectionsButton.setVisibility(View.VISIBLE);
+            }
+        });
+    }
+
+    @JavascriptInterface
+    public void hideGetDirectionsButton(){
+        webView.post(new Runnable() {
+            @Override
+            public void run() {
+                getDirectionsButton.setVisibility(View.GONE);
+            }
+        });
+    }
+    @JavascriptInterface
+    public String getPatientHomeDetailsForMapping(){
+        JSONObject personAddressObject = new JSONObject();
+        PersonAddress personAddress = patient.getPreferredAddress();
+        if(personAddress != null) {
+            try {
+                personAddressObject.put("longitude", personAddress.getLongitude());
+                personAddressObject.put("latitude", personAddress.getLatitude());
+                personAddressObject.put("patientSummary", patient.getDisplayName());
+            } catch (JSONException e) {
+                Log.e(getClass().getSimpleName(), "Could not get home location", e);
+            }
+        }
+        return personAddressObject.toString();
+    }
+
+    @JavascriptInterface
+    public String getMapsAPIKey(){
+        try {
+            MuzimaSetting muzimaSetting = ((MuzimaApplication) getApplicationContext()).getMuzimaSettingController()
+                    .getSettingByProperty(Constants.ServerSettings.MAPS_API_KEY);
+            if(muzimaSetting != null){
+                return muzimaSetting.getValueString();
+            }
+        } catch (MuzimaSettingController.MuzimaSettingFetchException e) {
+            Log.e(getClass().getSimpleName(), "Could not obtain API key",e);
+        }
+        return null;
+    }
+
+    private void initializeHomeLocationMapView(){
+
+        webView = findViewById(R.id.webview);
+
+        webView.setWebChromeClient(createWebChromeClient());
+        WebSettings webSettings = webView.getSettings();
+        webSettings.setRenderPriority(WebSettings.RenderPriority.HIGH);
+        webSettings.setJavaScriptEnabled(true);
+        webSettings.setDatabaseEnabled(true);
+        webSettings.setDomStorageEnabled(true);
+        webSettings.setBuiltInZoomControls(false);
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
+            webView.setWebContentsDebuggingEnabled(true);
+        }
+        webView.addJavascriptInterface(this,"patientLocationMapInterface");
+        webView.loadUrl("file:///android_asset/www/maps/patientHomeLocationMap.html");
+    }
+
+
+    private WebChromeClient createWebChromeClient() {
+        return new WebChromeClient() {
+            @Override
+            public void onProgressChanged(WebView view, int progress) {
+            }
+
+            @Override
+            public boolean onConsoleMessage(ConsoleMessage consoleMessage) {
+                String message = format("Javascript Log. Message: {0}, lineNumber: {1}, sourceId, {2}", consoleMessage.message( ),
```

Comment:
remove logs. Why is this method always returning true

Reply:
The logging here is fine. This method overrides one from ChromeClientWebview and is expected to return true if the ConsoleMessage was handled.